### PR TITLE
Feature: (ISA-192) Generate survey graphs

### PR DESCRIPTION
**Pull request by JoshGx2000**
Retrieves survey data from the database, and then generates graphs showing the number of surveys conducted over time.
Generates graphs showing data for all years, and only data post-2000.
South-west (2000 onwards)

Generated graphs stored in media directory.

### CODE REVIEW

File: backend/catalogue/management/commands/generate_survey_graphs.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    filename = title
+
+    if exclude_old_data:
+        min_year = max(min_year, 2000)
+        title = title + ' (2000 onwards)'
+
+    years = [v for v in range(min_year, max_year + 1)]
+
+    imagery = [
+        next((v.get('imagery_count') for v in data if v.get('year') == year), 0)
```
Comment (markhepburn) [inline backend/catalogue/management/commands/generate_survey_graphs.py:46]: I'm trying to nut this out: do you have duplicate years?  (It just seems a bit weird to be creating an iterator like that and calling next() on it.  I can see that it works.. just wondering if there's a better or clearer way).

Can you sort `data` by `year`, or even just turn data into a dict indexed by year?  (Now I'm needlessly looking for efficiency and away from clarity, but I bet there's an itertools function that would turn `[dict(video,sediment,etc)]` into `[[video],[sediment],..]`!)

Comment (markhepburn) [reply backend/catalogue/management/commands/generate_survey_graphs.py:46]: (It's basically a pivot, isn't it!  Turn a list of dicts with known keys and singular vals, into a dict of those keys and lists of vals.  Handling missing years might get tricky.  Anyway I'm rambling now)

File: backend/catalogue/management/commands/generate_survey_graphs.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+from django.core.management.base import BaseCommand
+from django.core.files.storage import default_storage
+from django.core.files import File
+from django.db import connections
+from collections import namedtuple
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MaxNLocator
+from io import BytesIO
```
Comment (markhepburn) [inline backend/catalogue/management/commands/generate_survey_graphs.py:8]: Probably a useful habit to get into by including logging (a chat I've been having with Oli lately)

**Review by markhepburn — COMMENTED** [review]
Looks good, but might be worth talking through a couple of the points in there.

File: backend/catalogue/management/commands/generate_survey_graphs.py (hunk 1 above)
Comment (JoshGx2000) [reply backend/catalogue/management/commands/generate_survey_graphs.py]: As discussed in Google Chat, this was just me forgetting stuff and being dumb. I've fixed this now so there's _way_ less redundant code, and the readability has greatly improved.

File: backend/catalogue/management/commands/generate_survey_graphs.py (hunk 2 above)
Comment (JoshGx2000) [reply backend/catalogue/management/commands/generate_survey_graphs.py:8]: Done!

**Pull request merged**